axon-framework/events/event-versioning.md

```diff
@@ -85,61 +85,120 @@ public class ComplaintEvent {
 Upcaster from 1.0 revision to 2.0 revision:
 
 ```java
-// Upcaster from 1.0 revision to 2.0 revision
-public class ComplaintEventUpcaster extends SingleEventUpcaster {
-
-    private static SimpleSerializedType targetType = 
-           new SimpleSerializedType(ComplainEvent.class.getTypeName(), "1.0");
-
-    @Override
-    protected boolean canUpcast(IntermediateEventRepresentation 
-                                    intermediateRepresentation) {
-        return intermediateRepresentation.getType().equals(targetType);
-    }
-
-    @Override
-    protected IntermediateEventRepresentation doUpcast(
-                        IntermediateEventRepresentation intermediateRepresentation) {
-        return intermediateRepresentation.upcastPayload(
-                new SimpleSerializedType(targetType.getName(), "2.0"),
-                org.dom4j.Document.class,
-                document -> {
-                    document.getRootElement()
-                            .addElement("description")
-                            .setText("no complaint description"); // Default value
-                    return document;
-                }
-        );
-    }
+public class ComplaintEvent1_to_2Upcaster extends SingleEventUpcaster {
```

Comment:
Can we add tabs? This one for jackson/json and another for xstream/xml

Reply:
Awesome plan! Gonna do that.